File: Chapter 06/6.4.3 The Modern Architect's AI Rule/Java/src/main/java/com/aiapi/demo/dtos/ShippingRequest.java
package com.aiapi.demo.dtos;

import io.swagger.v3.oas.annotations.media.Schema;

public class ShippingRequest {
    @Schema(description = "The unique ID of the physical product. Do NOT send digital product IDs.")
    private String productId;

    @Schema(description = "The destination zip code. Must be exactly 5 digits.")
    private String zipCode;

    // Standard Getters and Setters omitted for brevity
    public String getProductId() { return productId; }
    public void setProductId(String productId) { this.productId = productId; }
}

File: Chapter 06/Python/section_6_4_3_the_modern_architect_ai_rule/schema/shipping_request.py
from pydantic import BaseModel, Field

class ShippingRequest(BaseModel):
    """
    THE DTO (The AI's Input Form).
    Extracted from the controller to ensure proper Separation of Concerns.
    """
    product_id: str = Field(
        ..., 
        description="The unique ID of the physical product. Do NOT send digital product IDs (like MP3s or eBooks)."
    )
    zip_code: str = Field(
        ..., 
        description="The destination zip code. Must be exactly 5 digits."
    )